pkg/libvmi/status/status.go

```diff
@@ -133,3 +133,35 @@ func WithInterfaceStatus(interfaceStatus v1.VirtualMachineInstanceNetworkInterfa
 		vmiStatus.Interfaces = append(vmiStatus.Interfaces, interfaceStatus)
 	}
 }
+
+type VMOption func(vmiStatus *v1.VirtualMachineStatus)
+
+// WithStatus sets the status with specified value
+func WithVMStatus(status v1.VirtualMachineStatus) libvmi.VMOption {
+	return func(vm *v1.VirtualMachine) {
+		vm.Status = status
+	}
+}
+
+// New instantiates a new VM status configuration,
+// building its properties based on the specified With* options.
+func NewVMStatus(opts ...VMOption) v1.VirtualMachineStatus {
+	vmStatus := &v1.VirtualMachineStatus{}
+	for _, f := range opts {
+		f(vmStatus)
+	}
+
+	return *vmStatus
+}
+
+func WithVolumeUpdateState(volumeUpdateState *v1.VolumeUpdateState) VMOption {
```

Comment:
Currently it is not present but in future we could be add the `WithVolumeUpdateState` for vmi.
I would stay with using `VM` for VM related options
```suggestion
func WithVMVolumeUpdateState(volumeUpdateState *v1.VolumeUpdateState) VMOption {
```